openpaas-james/jmap/extensions/src/main/scala/com/linagora/openpaas/james/jmap/method/FilterSetMethod.scala

```diff
@@ -120,8 +121,8 @@ class FilterSetMethod @Inject()(val metricFactory: MetricFactory,
       .map(updateResult => updateResult.asFilterSetUpdateResults)
       .reduceWith(() => FilterSetUpdateResults.empty(), FilterSetUpdateResults.merge)
 
-  def updateRules(username: Username, validatedRules: List[Rule]): SMono[FilterSetUpdateResult] = {
-      SMono(filteringManagement.defineRulesForUser(username, validatedRules.asJava))
+  def updateRules(username: Username, validatedRules: List[Rule], ifInState: Option[FilterState]): SMono[FilterSetUpdateResult] = {
+      SMono(filteringManagement.defineRulesForUser(username, validatedRules.asJava, ifInState.map(filterState => new Version(filterState.state.toInt)).toJava))
```

Comment:
Extract `ifInState.map(filterState => new Version(filterState.state.toInt)).toJava` in a separate method, it is too technical here.